programs/libreplex_inscriptions/src/instructions/ghost/create_ghost_root_inscription.rs

```diff
@@ -0,0 +1,156 @@
+    pub validation_hash: Option<String>,
+    pub root: Pubkey,
+}
+
+
+impl CreateGhostRootInscriptionInput {
+    pub fn get_size(&self) -> usize {
+            1
+            + match self.authority {
+                Some(_) => 32,
+                None => 0,
+            } 
+            + 2 // default media type length
+            + 1 + match &self.validation_hash {
+                Some(x)=> x.len() + 4,
+                None => 0
+            }
+    }
+}
+
+/*
+   This endpoint is identical to v3 create.
+   However it only accepts LegacyMetadataSigners
+   and the root is passed in as an argument.
+
+   This is for roots that cannot exist as accounts on chain.
+*/
+
+const INITIAL_SIZE: usize = 8;
+#[derive(Accounts)]
+#[instruction(inscription_input: CreateGhostRootInscriptionInput)]
+pub struct CreateGhostRootInscription<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    // this must be either the root or else a PDA
+    // that is accepted as a proxy
+    #[account(mut)]
+    pub signer: Signer<'info>,
+
+    #[account(init_if_needed, seeds = [b"inscription_summary"],
+        bump, payer = payer, space = InscriptionSummary::BASE_SIZE)]
+    pub inscription_summary: Box<Account<'info, InscriptionSummary>>,
+
+    /*
+        generated as a PDA to make sure that each chunk address
+        can be derived from the base address
+    */
+    /// CHECK: Created outside and zero
+    #[account(
+        init,
+        payer = payer,
+        // starts with base anchor discriminator, although this will be overwritten
+        // by data
+        space = INITIAL_SIZE, // set a base rent for now, reduce later
+        seeds=[
+            "inscription_data".as_bytes(),
+            inscription_input.root.as_ref()
+        ],
+        bump)]
+    pub inscription_data: Account<'info, InscriptionData>,
+
+    /// CHECK: validated in logic
+    #[account(init,
+        space = Inscription::BASE_SIZE + inscription_input.get_size(),
+        seeds=[
+            "inscription_v3".as_bytes(),
+            inscription_input.root.as_ref()
+        ],
+        bump,
+        payer = payer)]
+    pub inscription_v3: Account<'info, InscriptionV3>,
+
+    pub system_program: Program<'info, System>,
+}
+
+pub fn handler(ctx: Context<CreateGhostRootInscription>, input: CreateGhostRootInscriptionInput) -> Result<()> {
+    let inscription_v3 = &mut ctx.accounts.inscription_v3;
+    let inscription_summary = &mut ctx.accounts.inscription_summary;
+
+    let authority = match input.authority {
+        Some(x) => x.to_owned(),
+        None => ctx.accounts.payer.key(),
+    };
+    let inscription_data = &ctx.accounts.inscription_data;
+
+    let clock = Clock::get()?;
+
+    inscription_summary.last_inscription_create_time = clock.unix_timestamp;
+    inscription_summary.last_inscription = inscription_v3.key();
+    inscription_summary.last_inscriber = ctx.accounts.payer.key();
+    inscription_summary.inscription_count_total += 1;
```

Comment:
let's refactor all this out into a separate method as discussed. so there's no duplication of logic between traditional and ghost mints